scania-digital-design-system/tegel
# feat(popover-core):  prevent internal logic

## **Describe pull-request**  
The user has initially reported this as an error, but digging further, I think this is a feature with a small fix.

Small fix:
 - Currently, we have an issue with the component's lifecycle methods. The component is to be updated before it is initialized, resulting in console.log errors. It can be seen in demo pages too on the form page. 
  - It was fixed by moving the initialization to the connectedCallback.

Feature:
Digging into the users issue recreated on [Stackblitz example](https://stackblitz.com/edit/stackblitz-starters-kvkg2u?file=src%2Fcomponents%2Fbict-popover-menu.component.ts) it was clear that user is taking full control over when a component should appear and there is no real use of `selector`or `referenceEl` however, component expect those to position itself. I have changed the core so that if user has not passed any of those props component will disable internal logic.

## **Issue Linking:**  
- **Jira:** Add ticket number after `CDEP-`: [CDEP-3532](https://tegel.atlassian.net/browse/CDEP-3532)    
- **GitHub:** https://github.com/scania-digital-design-system/tegel/issues/790


## **How to test**  
1. Try example that user has send, observe if you get errors in console. 
2. Try installing this [beta version](https://www.npmjs.com/package/@scania/tegel-angular-17/v/1.18.0-popover-menu-beta.4) in the same example and set disableLogic prop to true on tds-popover-menu.
3. Try installing [this beta](https://www.npmjs.com/package/@scania/tegel-angular-17/v/1.18.0-popover-menu-beta.4) version in the same example and remove selector and referenceEl prop from tds-popover-menu.
4. If no2 and no3 work without posting errors in console.log, we are good to go.

## **Checklist before submission**

## **Suggested test steps**




[CDEP-3532]: https://tegel.atlassian.net/browse/CDEP-3532?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
